### PR TITLE
port to es6

### PR DESCRIPTION
**Pull request by igord**

### CODE REVIEW

**Comment by jimmywarting** [comment]
I have been thinking of feature detecting first and then construct the saveAs method with what is avalible instead of trying everything in the same flow

> creating conditional function depending in javascript


```js
// instead of doing
 saveAs = function(...args) {

  if (supportsSaveAndOpen()) {
    use_IE_solution(args);
  } else if ( supportsAdownload() ) {
    use_a_download();
  } else {
    use_file_reader(args)
  }
}

// You can do
if (supportsSaveAndOpen() ) {
  saveAs = use_IE_solution
} else if (supportsAdownload() ) {
  saveAs = use_a_download
} else if (supportsFileReader()){
  saveAs = use_file_reader
} else {
  saveAs = () => { "can't save :(" }
}
```

That way you can garbage collect the other bloated code and the function will be more clean

**Comment by igord** [comment]
You are right, no need to create objectUrl when is not needed and I updated code. It is not difficult to separate conditional functions. I personally prefer it as it is.

**Comment by eligrey** [comment]
This may also need build system config data for transpiling to older supported browsers.

**Comment by jimmywarting** [comment]
Think this one is a bit overkill with promises, IE don't have it so it needs to be polyfilled

File: scripts/a-download.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/* global MouseEvent */
+
+const click = node => {
+  const event = new MouseEvent('click');
+  node.dispatchEvent(event);
+};
+
+export default ({ view, name, objectUrl }) => new Promise(resolve => {
+  const anchor = view.document.createElementNS('http://www.w3.org/1999/xhtml', 'a');
```
Comment (ogonkov) [inline scripts/a-download.js]: Each download will generate new DOM Anchor element? That's works different in current version

File: scripts/a-download.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  if (anchor.download === undefined) resolve(false);
+  setTimeout(() => {
```
Comment (ogonkov) [inline scripts/a-download.js]: What the reason to have set timeout inside Promise?

**Pull request closed without merging**